src/core/core/rpc_engine.cpp

```diff
@@ -154,29 +154,16 @@ namespace dsn {
             // server address side effect
             auto req = call->get_request();
             auto sp = task_spec::get(req->local_rpc_code);
-            if (reply->from_address != req->to_address)
+            // TODO(qinzuoyan): because current rpc forward is fake (resend from client), we cannot
```

Comment:
(1) glad to hear what is the exceptional case?
(2) This is too specific - previously the code structure is trying to handle group address in a systematic way instead of patching.